keymap.c

```diff
@@ -1498,7 +1498,8 @@ static enum CommandResult dump_bind_macro(struct Buffer *buf, struct Buffer *s,
   struct PagerData pdata = { 0 };
   struct PagerView pview = { &pdata };
 
-  pdata.fname = tempfile;
+  pdata.fname = buf_strdup(tempfile);
+  buf_pool_release(&tempfile);
```

Comment:
`buf_string()`, then `buf_pool_release()` after `mutt_do_pager()`

Reply:
should be fixed now